CMakeLists.txt

```diff
@@ -38,15 +38,51 @@ if ( ${SANITIZE} )
     add_link_options( -fsanitize=undefined,address -fno-omit-frame-pointer )
 endif ()
 
-# Quality of life external libs
-add_subdirectory ( 3rd-party/range-v3 EXCLUDE_FROM_ALL )
-add_subdirectory ( 3rd-party/spdlog EXCLUDE_FROM_ALL )
+function ( add_spirv_shader TARGET_NAME INPUT_FILE )
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv
+        COMMAND ${glslc} -c ${INPUT_FILE} -o ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv
+        MAIN_DEPENDENCY ${INPUT_FILE}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    )
+    add_custom_target ( ${TARGET_NAME} ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv )
+    set ( ${TARGET_NAME}_OUTPUTS ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv PARENT_SCOPE )
+endfunction ()
+
+function ( suppress_warnings TARGET_NAME )
+    get_target_property( target_options ${TARGET_NAME} COMPILE_OPTIONS )
+
+    if ( MSVC )
+        list( REMOVE_ITEM target_options /W4 )
+    else ()
+        list( REMOVE_ITEM target_options -Wall -Wextra -Wpedantic )
+    endif ()
+
+    set_property( TARGET ${TARGET_NAME} PROPERTY COMPILE_OPTIONS ${target_options} )
+endfunction ()
 
 # External dependecies
 find_package ( glfw3 REQUIRED )
 find_package ( Vulkan 1.3 REQUIRED )
 find_program ( glslc NAMES glslc HINTS Vulkan::glslc REQUIRED )
 
+# Quality of life external libs
+add_subdirectory ( 3rd-party/range-v3 EXCLUDE_FROM_ALL )
+add_subdirectory ( 3rd-party/spdlog EXCLUDE_FROM_ALL )
+
+# -v- VulkanMemoryAllocator configuration -v-
+set( VMA_STATIC_VULKAN_FUNCTIONS OFF )
+set( VMA_DYNAMIC_VULKAN_FUNCTIONS ON )
+
+# Force CMake to override lib's options with ours. 
+set( CMAKE_POLICY_DEFAULT_CMP0077 NEW )
```

Comment:
This policy required CMake 3.13 at least. So the cmake_minimum_version should get bumped accordingly

Reply:
Agree.